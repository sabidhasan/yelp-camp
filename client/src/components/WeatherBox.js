import React from 'react'

const weatherIcons = {
  'clear-day' : '☀️',
  'clear-night' : '☀️',
  'rain' : '☔️',
  'snow' : '⛄',
  'sleet' : '⛄',
  'wind' : '💨',
  'fog' : '🌁',
  'cloudy' : '☁️',
  'partly-cloudy-day' : '⛅',
  'partly-cloudy-night' : '⛅'
}

const WeatherBox = (props) => {
  if (!props.weather || !Object.keys(props.weather).length) return (
    <div className='WeatherBox'>
      <h2>No weather available</h2>
    </div>
  )

  return (
    <div className='WeatherBox'>
      <h2 className='WeatherBox__header'>Current Conditions</h2>
      <span className='WeatherBox__summary'>
        {weatherIcons[props.weather.icon]}
      </span>
      <span className='WeatherBox__text'>
        {props.weather.summary}
      </span>
      <span className='WeatherBox__item'>
        Temp {Math.round(props.weather.temperature)}℉
      </span>
      <span className='WeatherBox__item'>
        Humidity {Math.round(props.weather.humidity * 100)}%
      </span>
      <span className='WeatherBox__item'>
        Wind {Math.round(props.weather.windSpeed)} mph
      </span>
    </div>
  )
}


export default WeatherBox
