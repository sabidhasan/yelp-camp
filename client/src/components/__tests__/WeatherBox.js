import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import WeatherBox from '../WeatherBox'

configure({ adapter: new Adapter() });

describe('<WeatherBox />', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      weather: {
        icon: '',
        summary: 'Sample Summary',
        temperature: 50,
        humidity: 50,
        windSpeed: 5
      }
    }
    component = shallow(<WeatherBox { ...props } />)
  });

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('If no weather data, then render no data', () => {
    props = {
      weather: undefined
    }
    component = shallow(<WeatherBox { ...props } />)
    expect(component.find('.WeatherBox__error').text()).toBe('No weather available')
  });

  it('LoginForm component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  });  
});
