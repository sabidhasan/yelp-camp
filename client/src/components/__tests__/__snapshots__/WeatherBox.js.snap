// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherBox /> LoginForm component snapshot matches what is expected 1`] = `
<div
  className="WeatherBox"
>
  <h2
    className="WeatherBox__header"
  >
    Current Conditions
  </h2>
  <span
    aria-hidden="true"
    className="WeatherBox__summary"
  />
  <span
    className="WeatherBox__text"
  >
    Sample Summary
  </span>
  <span
    className="WeatherBox__item"
  >
    Temp 
    50
    â„‰
  </span>
  <span
    className="WeatherBox__item"
  >
    Humidity 
    5000
    %
  </span>
  <span
    className="WeatherBox__item"
  >
    Wind 
    5
     mph
  </span>
</div>
`;
